cloud/scope/machine_pool.go

```diff
@@ -342,89 +339,146 @@ func (m *MachinePoolScope) GetFreeFormTags() map[string]string {
 // ReconcileInstanceConfiguration works to try to reconcile the state of the instance configuration for the cluster
 func (m *MachinePoolScope) ReconcileInstanceConfiguration(ctx context.Context) error {
 	var instanceConfiguration *core.InstanceConfiguration
-	instanceConfigurationId := m.GetInstanceConfigurationId()
-
-	if len(instanceConfigurationId) > 0 {
-		req := core.GetInstanceConfigurationRequest{InstanceConfigurationId: common.String(instanceConfigurationId)}
-		instanceConfiguration, err := m.ComputeManagementClient.GetInstanceConfiguration(ctx, req)
-		if err == nil {
-			m.Info("instance configuration found", "InstanceConfigurationId", instanceConfiguration.Id)
-			m.SetInstanceConfigurationIdStatus(instanceConfigurationId)
-			return m.PatchObject(ctx)
-		} else {
-			return errors.Wrap(err, fmt.Sprintf("error getting instance configuration by id %s", instanceConfigurationId))
+	instanceConfiguration, err := m.GetInstanceConfiguration(ctx)
+	if err != nil {
+		return err
+	}
+	freeFormTags := m.GetFreeFormTags()
+	definedTags := make(map[string]map[string]interface{})
+	if m.OCICluster.Spec.DefinedTags != nil {
+		for ns, mapNs := range m.OCICluster.Spec.DefinedTags {
```

Comment:
I know it is a short lived scope so no need to change but what does the N in `mapNs` stand for?